spring-ai-core/src/main/java/org/springframework/ai/model/Media.java

```diff
@@ -119,4 +145,199 @@ public String getId() {
 		return this.id;
 	}
 
+	public String getName() {
+		return this.name;
+	}
+
+	/**
+	 * Builder class for Media.
+	 */
+	public static class Builder {
+
+		private String id = MEDIA_NO_ID;
+
+		private MimeType mimeType;
+
+		private Object data;
+
+		private String name;
+
+		public Builder mimeType(MimeType mimeType) {
+			Assert.notNull(mimeType, "MimeType must not be null");
+			this.mimeType = mimeType;
+			return this;
+		}
+
+		public Builder data(Resource resource) {
+			Assert.notNull(resource, "Data must not be null");
+			try {
+				this.data = resource.getContentAsByteArray();
+			}
+			catch (IOException e) {
+				throw new IllegalArgumentException(e);
+			}
+			return this;
+		}
+
+		public Builder data(Object data) {
+			Assert.notNull(data, "Data must not be null");
+			this.data = data;
+			return this;
+		}
+
+		public Builder data(URL url) {
+			Assert.notNull(url, "URL must not be null");
+			this.data = url.toString();
+			;
+			return this;
+		}
+
+		public Builder id(String id) {
+			Assert.notNull(id, "Id must not be null");
+			this.id = id;
+			return this;
+		}
+
+		public Builder name(String name) {
+			Assert.notNull(name, "Name must not be null");
+			this.name = name;
+			return this;
+		}
+
+		public Media build() {
+			return new Media(mimeType, data, id, name);
```

Comment:
should verify that required fields are set, I presume `mimeType` and `media` are required but there can be defaults generated for `id` and `name`.  The default for no media name shouldn't be `nope` and it seems like `id` could be null?  It is retrieved from the model response for the Audio response.